examples/quickstart/main.rs

```diff
@@ -0,0 +1,124 @@
+use std::{collections::HashMap, env};
+
+use futures::{future::select_all, StreamExt};
+use num_bigint::BigUint;
+use tokio::{sync::mpsc, task, task::JoinHandle};
+use tracing_subscriber::EnvFilter;
+use tycho_simulation::{
+    evm::{
+        engine_db::tycho_db::PreCachedDB,
+        protocol::{
+            filters::balancer_pool_filter, uniswap_v2::state::UniswapV2State,
+            vm::state::EVMPoolState,
+        },
+        stream::ProtocolStreamBuilder,
+    },
+    models::Token,
+    protocol::models::BlockUpdate,
+    tycho_client::feed::component_tracker::ComponentFilter,
+    tycho_core::dto::Chain,
+    utils::load_all_tokens,
+};
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt()
+        .with_env_filter(EnvFilter::from_default_env())
+        .with_target(false)
+        .init();
+
+    let tycho_url =
+        env::var("TYCHO_URL").unwrap_or_else(|_| "tycho-beta.propellerheads.xyz".to_string());
+    let tycho_api_key: String =
+        env::var("TYCHO_API_KEY").unwrap_or_else(|_| "sampletoken".to_string());
+
+    let tvl_threshold = 10_000.0;
+
+    let all_tokens = load_all_tokens(tycho_url.as_str(), Some(tycho_api_key.as_str())).await;
+
+    // Communication channels
+    let (tick_tx, mut tick_rx) = mpsc::channel::<BlockUpdate>(12);
+
+    // Producer task: Fetch updates and send them via channel
+    let tycho_message_processor: JoinHandle<anyhow::Result<()>> = tokio::spawn(async move {
+        let tvl_filter = ComponentFilter::with_tvl_range(tvl_threshold, tvl_threshold);
+
+        let mut protocol_stream = ProtocolStreamBuilder::new(&tycho_url, Chain::Ethereum)
+            .exchange::<UniswapV2State>("uniswap_v2", tvl_filter.clone(), None)
+            .exchange::<EVMPoolState<PreCachedDB>>(
+                "vm:balancer_v2",
+                tvl_filter.clone(),
+                Some(balancer_pool_filter),
+            )
+            .auth_key(Some(tycho_api_key.clone()))
+            .set_tokens(all_tokens.clone())
+            .await
+            .build()
+            .await
+            .expect("Failed building protocol stream");
+
+        while let Some(msg) = protocol_stream.next().await {
+            tick_tx
+                .send(msg.unwrap())
+                .await
+                .expect("Sending tick failed!");
```

Comment:
Why not just do the printing here? I think having the channels makes the example unnecessarily complex.